src/uint.rs

```diff
@@ -335,6 +335,139 @@ macro_rules! impl_uint_aliases {
+    (U2368, 2368, "2368-bit"),
+    (U2432, 2432, "2432-bit"),
+    (U2496, 2496, "2496-bit"),
+    (U2560, 2560, "2560-bit"),
+    (U2624, 2624, "2624-bit"),
+    (U2688, 2688, "2688-bit"),
+    (U2752, 2752, "2752-bit"),
+    (U2816, 2816, "2816-bit"),
+    (U2880, 2880, "2880-bit"),
+    (U2944, 2944, "2944-bit"),
+    (U3008, 3008, "3008-bit"),
+    (U3072, 3072, "3072-bit"),
+    (U3136, 3136, "3136-bit"),
+    (U3200, 3200, "3200-bit"),
+    (U3264, 3264, "3264-bit"),
+    (U3328, 3328, "3328-bit"),
+    (U3392, 3392, "3392-bit"),
+    (U3456, 3456, "3456-bit"),
+    (U3520, 3520, "3520-bit"),
+    (U3584, 3584, "3584-bit"),
+    (U3648, 3648, "3648-bit"),
+    (U3712, 3712, "3712-bit"),
+    (U3776, 3776, "3776-bit"),
+    (U3840, 3840, "3840-bit"),
+    (U3904, 3904, "3904-bit"),
+    (U3968, 3968, "3968-bit"),
+    (U4032, 4032, "4032-bit"),
+    (U4096, 4096, "4096-bit"),
+    (U4160, 4160, "4160-bit"),
+    (U4224, 4224, "4224-bit"),
+    (U4288, 4288, "4288-bit"),
+    (U4352, 4352, "4352-bit"),
+    (U4416, 4416, "4416-bit"),
+    (U4480, 4480, "4480-bit"),
+    (U4544, 4544, "4544-bit"),
+    (U4608, 4608, "4608-bit"),
+    (U4672, 4672, "4672-bit"),
+    (U4736, 4736, "4736-bit"),
+    (U4800, 4800, "4800-bit"),
+    (U4864, 4864, "4864-bit"),
+    (U4928, 4928, "4928-bit"),
+    (U4992, 4992, "4992-bit"),
+    (U5056, 5056, "5056-bit"),
+    (U5120, 5120, "5120-bit"),
+    (U5184, 5184, "5184-bit"),
+    (U5248, 5248, "5248-bit"),
+    (U5312, 5312, "5312-bit"),
+    (U5376, 5376, "5376-bit"),
+    (U5440, 5440, "5440-bit"),
+    (U5504, 5504, "5504-bit"),
+    (U5568, 5568, "5568-bit"),
+    (U5632, 5632, "5632-bit"),
+    (U5696, 5696, "5696-bit"),
+    (U5760, 5760, "5760-bit"),
+    (U5824, 5824, "5824-bit"),
+    (U5888, 5888, "5888-bit"),
+    (U5952, 5952, "5952-bit"),
+    (U6016, 6016, "6016-bit"),
+    (U6080, 6080, "6080-bit"),
+    (U6144, 6144, "6144-bit"),
+    (U6208, 6208, "6208-bit"),
+    (U6272, 6272, "6272-bit"),
+    (U6336, 6336, "6336-bit"),
+    (U6400, 6400, "6400-bit"),
+    (U6464, 6464, "6464-bit"),
+    (U6528, 6528, "6528-bit"),
+    (U6592, 6592, "6592-bit"),
+    (U6656, 6656, "6656-bit"),
+    (U6720, 6720, "6720-bit"),
+    (U6784, 6784, "6784-bit"),
+    (U6848, 6848, "6848-bit"),
+    (U6912, 6912, "6912-bit"),
+    (U6976, 6976, "6976-bit"),
+    (U7040, 7040, "7040-bit"),
+    (U7104, 7104, "7104-bit"),
+    (U7168, 7168, "7168-bit"),
+    (U7232, 7232, "7232-bit"),
+    (U7296, 7296, "7296-bit"),
+    (U7360, 7360, "7360-bit"),
+    (U7424, 7424, "7424-bit"),
+    (U7488, 7488, "7488-bit"),
+    (U7552, 7552, "7552-bit"),
+    (U7616, 7616, "7616-bit"),
+    (U7680, 7680, "7680-bit"),
+    (U7744, 7744, "7744-bit"),
+    (U7808, 7808, "7808-bit"),
+    (U7872, 7872, "7872-bit"),
+    (U7936, 7936, "7936-bit"),
+    (U8000, 8000, "8000-bit"),
+    (U8064, 8064, "8064-bit"),
+    (U8128, 8128, "8128-bit"),
+    (U8192, 8192, "8192-bit")
+}
+
+#[cfg(not(feature = "extra-sizes"))]
```

Comment:
Instead of having a `not`, `extra-sizes` could be purely additive.

That would also take care of the `doc_auto_cfg` documentation, so the extra sizes are appropriately tagged as needing the feature.

Reply:
Problem is, the original set of supported sizes is large and looks abit "random" IMO. If, e.g., we'd have `U64, U128, U256, U512, U1024, U2048, U4096, U8192` as the types supported by default and everything else requiring `extra_sizes`, this would make sense. However, I don't think we can do that can we? (I'm not sure if feature-gating a non-feature gated type is considered a break of SemVer or not)

Otherwise, with the way things are currently, this becomes a little grotesque. However, if you'd still prefer that, I'll make the change. Remember however, that these distinctions will become harder with the `extra_trait_impls` macros.

Reply:
(Your suggestion was my original line of thought and I chose against it because of the aforementioned reasoning)

Reply:
I think one way or another the types which require a feature need to be annotated with the feature name in the rustdoc.

If you simply define them in a module which is feature-gated, `doc_auto_cfg` should take care of that automatically.

Reply:
Can I keep only `U64, U128, U256, U512, U1024, U2048, U4096, U8192` outside the feature-gate, or should I stick with the current set?

Reply:
Should be good now